app/src/main/java/com/example/happymeals/Login.java

```diff
@@ -0,0 +1,114 @@
+package com.example.happymeals;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.happymeals.database.FireAuth;
+
+/** Initial UI layer for user login
+ * This class is a candidate to be the entry point of the application -- its implementation has yet
+ *  to be approved. It features a login screen and will authenticate users. Upon successful authentication,
+ *  users can then access all their persistent data.
+ * @author sruduke bfiogbe
+ *
+ */
+
+
+public class Login extends AppCompatActivity {
+
+    /**
+     * This is the function called whenever the LoginScreen is created -- in our
+     * case, this is on the launch of the app (if approved by client). If implemented,
+     * upon successful user authentication, it will send an intent to open an instance of
+     * the MainActivity. From there, users will be able to access all of their data -- ingredients,
+     * recipes, meal plans, etc.
+     * @param savedInstanceState The instance state to restore the activity to (if applicable) {@link Bundle}
+     */
+
+
+    private Button login, register;
```

Comment:
These names are very ambiguous, consider changing them to something like loginButton, or forgotPasswordLabel, etc.